### PR TITLE
[SYCL][Graph][Doc] Add SYCL-Graph usage guide and example doc

### PR DESCRIPTION
**Pull request by Bensuo**
- Create new `syclgraph` folder in `doc` for syclgraph related, user-facing documents
- Add a new usage guide document with some usage guidelines for common scenarios
- Move examples from spec to usage guide to reduce bloat
- Remove reductions from examples since they are not supported

### CODE REVIEW

File: sycl/doc/syclgraph/SYCLGraphUsageGuide.md [hunk 1]
```diff
@@ -0,0 +1,515 @@
...
+Examples here will generally only use one graph creation API ("Explicit" or
+"Record & Replay") for simplicity, but can generally be applied to either API
+unless specifically noted.
+
+### Use Host-Tasks For Host Work
+
+SYCL Graph cannot capture work done on the host in application code which does
+not go through the SYCL runtime, including direct operations (for example kernel
+submissions) through a backend API (level-zero, cuda, etc). Any work to be done
+on the host should be captured within a `sycl::host-task` as a node in the
```
Comment (EwanC) [inline sycl/doc/syclgraph/SYCLGraphUsageGuide.md:24]:
```suggestion
on the host should be captured within a `sycl::host_task` as a node in the
```

or maybe `handler::host_task`

File: sycl/doc/syclgraph/SYCLGraphUsageGuide.md [hunk 2]
```diff
@@ -0,0 +1,515 @@
...
+    do_some_host_work();
+    CGH.parallel_for(...);
+});
+```
+
+In normal SYCL usage `do_some_host_work()` will be evaluated during the call to
+submit. With SYCL Graph this code will be evaluated once during the call to
+`submit()` or `command_graph::add()` but will not be evaluated on subsequent
+runs of the graph which may lead to incorrect behavior. This code should be
+placed into a `host-task` node which the kernel (or other operation) depends on,
```
Comment (EwanC) [inline sycl/doc/syclgraph/SYCLGraphUsageGuide.md:49]:
```suggestion
placed into a `host_task` node which the kernel (or other operation) depends on,
```

**Review by EwanC — APPROVED** [review]

File: sycl/doc/syclgraph/SYCLGraphUsageGuide.md (hunk 1 above)
Comment (Bensuo) [reply sycl/doc/syclgraph/SYCLGraphUsageGuide.md]: I've removed the monospace from this usage and just gone with "host-task" to be consistent with other usage, and I don't think the monospace is adding anything here.

File: sycl/doc/syclgraph/SYCLGraphUsageGuide.md (hunk 2 above)
Comment (Bensuo) [reply sycl/doc/syclgraph/SYCLGraphUsageGuide.md]: Changed to just host-task as per https://github.com/intel/llvm/pull/14965#discussion_r1705376638

**Review by reble — APPROVED** [review]

**Review by AlexeySachkov — APPROVED** [review]

**Comment by Bensuo** [comment]
@intel/llvm-gatekeepers This is ready to merge, thanks

**Pull request merged**